### PR TITLE
Fix form subtext for entity list forms

### LINKED ISSUE #6726: Subtext changes only in the first update form when the dataset is updated
#### ODK Collect version
the master version fd531039defffb3a084ae2704490e69a0672df94
#### Android version
10, 14
#### Device used
Redmi 9T, Pixel 7a
#### Problem description
After changes in #6712 a dataset is downloaded only once. The subtext in "Start new form" is changed only in the first update form, although all update forms were updated (the new entity is in the list).
The screenshot below shows an updated dataset and different subtexts (time) in update forms.

Image

#### Steps to reproduce the problem
1. Download a project with more than one update form e.g. user "trees" https://test.getodk.cloud/projects/621/app-users
2. In Central create a new entity.
3. Refresh the list of blank forms in Collect.
4. Check the subtext under each update form in "Start new form".
#### Expected behavior
The subtext should be updated.

### PR DESCRIPTION
**Pull request by seadowg**
Closes https://github.com/getodk/collect/issues/6726

#### Why is this the best possible solution? Were any other approaches considered?

I think down the line we're most likely going to want to have some explicit link between entity lists and forms, but it feels like designing that should be held off on until we're looking at things like "entity first" UI. I'd be worried otherwise, we'd end up with something we'd have to make significant changes to.

For now, adding a "last updated" timestamp to entity lists provides a fairly simple way to work out when entity lists are updated from the perspective of a form (as it can be compared to the forms added/updated date).

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

The key areas here are to check that forms with updated entity lists display correctly in form lists (except for the cases we're not covering laid out [here](https://github.com/getodk/collect/issues/6726#issuecomment-2909842078)). Also, it'd be good to make sure that functionality around blocking form entry during updates for entity forms works as expected as that logic is involved to the changes I had to make.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by seadowg** [comment]
Marking this as "needs testing" early as I'm worried there might be edge cases we need to do deal with here.

File: collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt [hunk 1]
```diff
@@ -16,11 +16,14 @@ import org.odk.collect.shared.Query
 
 abstract class EntitiesRepositoryTest {
 
-    abstract fun buildSubject(): EntitiesRepository
+    abstract fun buildSubject(clock: () -> Long): EntitiesRepository
+    fun buildSubject(): EntitiesRepository {
+        return buildSubject { 0 }
+    }
```
Comment (grzesiek2010) [inline collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt]: This could be just:
`abstract fun buildSubject(clock: () -> Long = { 0L }): EntitiesRepository`

File: collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormUseCases.kt [hunk 2]
```diff
@@ -109,13 +117,23 @@ object ServerFormUseCases {
...
-                        entitiesDownloaded = true
                     } catch (t: Throwable) {
                         throw EntityListUpdateException(t)
                     }
+                } else {
+                    val existingForm = formsRepository.getAllByFormIdAndVersion(formToDownload.formId, formToDownload.formVersion).getOrNull(0)
+                    if (existingForm != null) {
+                        val entityListLastUpdated = localEntityList.lastUpdated
+                        if (entityListLastUpdated != null) {
+                            if (entityListLastUpdated > existingForm.getLastUpdated()) {
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormUseCases.kt]: Wouldn't it be better to merge those two nested if statements?

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on a device with Android 10

**Verified cases:**
- issue #6726 doesn't occur
- updating after adding a new entity
- updating after adding a new property
- updating after deleting an form
- updating after adding a new version of a form
- blocking form entry during updates

**Review by grzesiek2010 — APPROVED** [review]

**Comment by WKobus** [comment]
**Tested with success**

Verified on device with Android 15
